crates/storage/src/collections/mapping/mod.rs

```diff
@@ -53,52 +52,49 @@ impl<K, V> core::fmt::Debug for Mapping<K, V> {
     }
 }
 
-impl<K, V> Mapping<K, V>
-where
-    K: PackedLayout,
-    V: PackedLayout,
-{
+impl<K, V> Mapping<K, V> {
     /// Creates a new empty `Mapping`.
-    ///
-    /// TODO [#961]: Ideally we improve how this is initialized by extending the
-    /// `SpreadLayout`/`PackedLayout` traits for non-caching data structures.
-    pub fn new(offset_key: Key) -> Self {
+    fn new(offset_key: Key) -> Self {
         Self {
             offset_key,
             _marker: Default::default(),
         }
     }
+}
 
+impl<K, V> Mapping<K, V>
+where
+    K: PackedLayout,
+    V: PackedLayout,
+{
     /// Insert the given `value` to the contract storage.
-    pub fn insert<Q, R>(&mut self, key: &Q, value: &R)
+    #[inline]
+    pub fn insert<Q, R>(&mut self, key: Q, value: &R)
     where
-        K: core::borrow::Borrow<Q>,
-        Q: scale::Encode,
-        V: core::borrow::Borrow<R>,
-        R: PackedLayout,
+        Q: scale::EncodeLike<K>,
```

Comment:
Can I think of `EncodeLike` sort-of like `EncodeLike = Borrow + Encode`?

Reply:
EncodeLike is a purely marker trait coming from the parity-scale-codec crate. Please go to the crates docs to read more about its general use.

Reply:
I read the docs, and I think my question is still valid 😄

I think the semantics of a type being able to be SCALE encoded as another type (`EncodeLike`) vs. being able to be borrowed as another type (`Borrow`) are slightly different, which is why I asked